docs/examples.md

```diff
@@ -0,0 +1,152 @@
+# Examples
+
+__Skipper__ provides different variations of `Set` and `Map` implementations with STL-like interface 
+which use __Skip List__ under the hood.
+
+## Sequential
+
+Sequential classes like [`SequentialSkipListSet`](../include/skipper/sequential_set.hpp) 
+and [`SequentialSkipListMap`](../include/skipper/sequential_map.hpp) are meant to be used in a single threaded
+environment, i.e. they are not thread-safe. 
+
+### Interface
+
+Both of them provide a subset of STL-like interface:
+```cpp
+template <typename T>
+class SequentialSkipListSet {
+ public:
+  // O(log N) complexity
+  auto Find(const T& value) const -> Iterator;
+  auto Insert(const T& value) -> std::pair<Iterator, bool>;
+  auto Erase(const T& value) -> std::size_t;
+
+  // O(1) complexity
+  auto Begin() const -> Iterator;
+  auto End() const -> Iterator;
+};
+
+template <typename Key, typename Value>
+class SequentialSkipListMap {
+ public:
+  // O(log N) complexity
+  auto Find(const Key& key) const -> Iterator;
+  auto Insert(const Key& key, const Value& value) -> std::pair<Iterator, bool>;
+  auto operator[](const Key& key) -> Value&;
+  auto operator[](const Key& key) const -> const Value&;
+  auto Erase(const Key& key) -> std::size_t;
+
+  // O(1) complexity
+  auto Begin() const -> Iterator;
+  auto End() const -> Iterator;
+};
+```
+
+### Iterator
+
+Note that `Iterator` is of Forward category (see [here](https://en.cppreference.com/w/cpp/iterator/forward_iterator)):
+```cpp
+template <typename T>
+class Iterator {
+ public:
+  // O(1) complexity
+  auto operator*() const -> const T&;
+  auto operator->() const -> const T*;
+  auto operator++(/* prefix */) -> Iterator&;
+  auto operator++(int /* postfix */) -> Iterator;
+  auto operator==(const Iterator& other) const -> bool;
+  auto operator!=(const Iterator& other) const -> bool;
+};
+```
```

Comment:
Add `SequentialSkipListMap` Iterator description as some operators (`operator*()` and `operator->()`) are slightly different.

Reply:
Sorry, fixed.